src/transformers/models/ernie_m/modeling_ernie_m.py

```diff
@@ -0,0 +1,1065 @@
+        self.encoder = ErnieMEncoder(config)
+        self.pooler = ErnieMPooler(config) if add_pooling_layer else None
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.embeddings.word_embeddings
+
+    def set_input_embeddings(self, value):
+        self.embeddings.word_embeddings = value
+
+    def _prune_heads(self, heads_to_prune):
+        """
+        Prunes heads of the model. heads_to_prune: dict of {layer_num: list of heads to prune in this layer} See base
+        class PreTrainedModel
+        """
+        for layer, heads in heads_to_prune.items():
+            self.encoder.layers[layer].self_attn.prune_heads(heads)
+
+    @add_start_docstrings_to_model_forward(ERNIE_M_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
+    @add_code_sample_docstrings(
+        processor_class=_TOKENIZER_FOR_DOC,
+        checkpoint=_CHECKPOINT_FOR_DOC,
+        output_type=BaseModelOutputWithPastAndCrossAttentions,
+        config_class=_CONFIG_FOR_DOC,
+    )
+    def forward(
+        self,
+        input_ids: Optional[tensor] = None,
+        position_ids: Optional[tensor] = None,
+        attention_mask: Optional[tensor] = None,
+        head_mask: Optional[tensor] = None,
+        inputs_embeds: Optional[tensor] = None,
+        past_key_values: Optional[Tuple[Tuple[tensor]]] = None,
+        use_cache: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ):
+        if input_ids is not None and inputs_embeds is not None:
+            raise ValueError("You cannot specify both input_ids and inputs_embeds at the same time.")
+
+        # init the default bool value
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+
+        head_mask = self.get_head_mask(head_mask, self.config.num_hidden_layers)
+
+        past_key_values_length = 0
+        if past_key_values is not None:
+            past_key_values_length = past_key_values[0][0].shape[2]
+
+        if attention_mask is None:
+            attention_mask = (input_ids == 0).to(torch.float32) * -1e4
+            if past_key_values is not None:
+                batch_size = past_key_values[0][0].shape[0]
+                past_mask = torch.zeros([batch_size, 1, 1, past_key_values_length], dtype=attention_mask.dtype)
+                attention_mask = torch.concat([past_mask, attention_mask], dim=-1)
+        # For 2D attention_mask from tokenizer
+        elif attention_mask.ndim == 2:
+            attention_mask = attention_mask.to(torch.float32)
+            attention_mask = (1.0 - attention_mask) * -1e4
```

Comment:
For all of the `minimum` values we use `torch.finfo(scores.dtype).min
```suggestion
        if attention_mask is None:
            attention_mask = (input_ids == 0).to(torch.float32) *torch.finfo(attention_mask.dtype).min
            if past_key_values is not None:
                batch_size = past_key_values[0][0].shape[0]
                past_mask = torch.zeros([batch_size, 1, 1, past_key_values_length], dtype=attention_mask.dtype)
                attention_mask = torch.concat([past_mask, attention_mask], dim=-1)
        # For 2D attention_mask from tokenizer
        elif attention_mask.ndim == 2:
            attention_mask = attention_mask.to(torch.float32)
            attention_mask = (1.0 - attention_mask) * torch.finfo(attention_mask.dtype).min
```

Reply:
Also why assume that the attention mask should be 1 where input ids is 0 ? We usually just init to torch.ones.

Reply:
+ the Is there a reason for not using [`get_extended_attention_mask`](https://github.com/ArthurZucker/transformers/blob/main/src/transformers/modeling_utils.py#L810)

Reply:
@ArthurZucker thanks for the comments! I was not getting the desired output with `get_extended_attention_mask` but it was all good with paddlenlp implementation so I stuck with that.

Reply:
I am guessing it is up to you. They should (theoretically ?) output the same, but if it is important that the default behavior is maintaned, keep it and add the `#copied from paddlenlp` that would be great (as a reference, maybe `# Adapted ` in case there are issues with copied

Reply:
Ok I will use `#adapted`, 


I didn't get this point, please elaborate - 
```
I think we should port the weights to the paddle paddle organization (and change the paths here) and its good to go!
```
the weights at [https://huggingface.co/PaddlePaddle/ernie-m-base](https://huggingface.co/PaddlePaddle/ernie-m-base) are paddle weights do you mean to push the pytorch weights at [https://huggingface.co/susnato/ernie-m-base_pytorch](https://huggingface.co/susnato/ernie-m-base_pytorch) to the official [paddle org repo](https://huggingface.co/PaddlePaddle)  to make it more reasonable?

Reply:
Oh sorry! I mean yeah we should transfer ownership of the weights, but we can do this later on